x/staking/keeper/msg_server.go

```diff
@@ -446,16 +444,15 @@ func (k msgServer) Undelegate(ctx context.Context, msg *types.MsgUndelegate) (*t
 		}()
 	}
 
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	sdkCtx.EventManager().EmitEvents(sdk.Events{
-		sdk.NewEvent(
-			types.EventTypeUnbond,
-			sdk.NewAttribute(types.AttributeKeyValidator, msg.ValidatorAddress),
-			sdk.NewAttribute(types.AttributeKeyDelegator, msg.DelegatorAddress),
-			sdk.NewAttribute(sdk.AttributeKeyAmount, undelegatedCoin.String()),
-			sdk.NewAttribute(types.AttributeKeyCompletionTime, completionTime.Format(time.RFC3339)),
-		),
-	})
+	if err := k.environment.EventService.EventManager(ctx).EmitKV(
```

Comment:
Would this be only be an EmitKV or could it also potentially also be an Emit too?

Reply:
it could be emit but since we dont have the types defined i went with EmitKV